sdk/core/core-http/lib/serviceClient.ts

```diff
@@ -447,6 +462,10 @@ function createDefaultRequestPolicyFactories(authPolicyFactory: RequestPolicyFac
     factories.push(throttlingRetryPolicy());
   }
 
+  if (logger.info.enabled) {
```

Comment:
I've moved this check into the policy, but the question still applies. We check for info because all the logging coming from this policy logs to the info channel. Really this is an optimization to avoid doing all the serialization work needed to log. Most of the time, if we're just logging strings, we can just log to info and the logging machinery will ensure it's not printed unless the log level is set appropriately.